llvm/test/SYCLLowerIR/SpecConstants/SYCL-alloca.ll

```diff
@@ -1,7 +1,6 @@
-; RUN: sycl-post-link -properties -split=auto -spec-const=native -S -o %t.table %s -generate-device-image-default-spec-consts
```

Comment:
Is the sycl-post-link test not expected to work anymore? I somehow feel it will be better to add a new test here.

Reply:
sycl-post-link test is still expected to work.
This test was only checking output of SpecConstantPass, while other tests in llvm/test/tools/sycl-post-link/ folder are also checking sycl-post-link tool output. So I move this test into llvm/test/SYCLLowerIR/SpecConstants/SYCL-alloca.ll

After reading your comment, I also feel it is better to keep the test to check sycl-post-link is correctly passing parameters to SpecConstantPass. So I added it back in the second commit.